elastic/kibana
# [Alerting] Add telemetry for number of scheduled actions during rule execution

Towards https://github.com/elastic/kibana/issues/122535

## Summary

Adds telemetry to calculate the p50/p90/p99 values for the number of actions a rule has scheduled during a single rule execution. Calculates by rule type and across all rule types.

## To Verify

1. Update the alerting telemetry task to run more frequently:

```
--- a/x-pack/plugins/alerting/server/usage/task.ts
+++ b/x-pack/plugins/alerting/server/usage/task.ts
@@ -139,7 +139,7 @@ export function telemetryTaskRunner(
                   avg_execution_time_per_day: dailyExecutionCounts.avgExecutionTime,
                   avg_execution_time_by_type_per_day: dailyExecutionCounts.avgExecutionTimeByType,
                 },
-                runAt: getNextMidnight(),
+                runAt: new Date(), // getNextMidnight(),
               };
             }
           )
```

2. Create some rules that schedule actions and let them run for a little
3. Navigate to `https://localhost:5601/api/stats?extended=true&legacy=true` and see the following fields inside alerting telemetry:
* `alerts.percentile_num_scheduled_actions_per_day`
* `alerts.percentile_num_scheduled_actions_by_type_per_day `

### Checklist
